### ISSUE TITLE
T4 VS Project Integration Error

### ISSUE THREAD

**Issue by dcastenholz** [msawczyn/EFDesigner#77]
When nested folders are specified for the targets of the t4 generated files, the top level folder will be created, but if there are multiple sub folders under the main folder, only the first one will be created.  This results in the newly generated files being added to the VS project under the xxx.tt template file.

Found the cause in MultipleOutputHelper.ttinclude.  Initialization of **bool found = false**; is in the wrong place.

**Please change:**

```
      private ProjectItem GetOrCreateParentItem(string filePath)
      {
         if (string.IsNullOrEmpty(filePath))
            return templateProjectItem;

         string projectDirectory = Path.GetDirectoryName(templateProjectItem.ContainingProject.FullName);
         string fileDirectory = Path.GetDirectoryName(filePath);

         if (fileDirectory.ToLower() == projectDirectory.ToLower())
            return templateProjectItem;

         ProjectItem result = templateProjectItem;

         string relativeFilePath = fileDirectory.Substring(projectDirectory.Length + 1);
         Queue<string> pathParts = new Queue<string>(relativeFilePath.Split('\\'));
         ProjectItems currentItemList = templateProjectItem.ContainingProject.ProjectItems;
=>       bool found = false;

         while (pathParts.Any())
         {
                  string pathPart = pathParts.Dequeue();
...
```

**To:**

```
      private ProjectItem GetOrCreateParentItem(string filePath)
      {
         if (string.IsNullOrEmpty(filePath))
            return templateProjectItem;

         string projectDirectory = Path.GetDirectoryName(templateProjectItem.ContainingProject.FullName);
         string fileDirectory = Path.GetDirectoryName(filePath);

         if (fileDirectory.ToLower() == projectDirectory.ToLower())
            return templateProjectItem;

         ProjectItem result = templateProjectItem;

         string relativeFilePath = fileDirectory.Substring(projectDirectory.Length + 1);
         Queue<string> pathParts = new Queue<string>(relativeFilePath.Split('\\'));
         ProjectItems currentItemList = templateProjectItem.ContainingProject.ProjectItems;

         while (pathParts.Any())
         {
=>                bool found = false;
                  string pathPart = pathParts.Dequeue();
...
```

**Comment by dcastenholz** [comment]
Note that this bug only occurs on the initial creation of the generated files that are in nested folders.  Once the folders are already added, the existing project integration works as designed.  The above code fix will correct the initial folder additions to the VS project.
To test for the bug, create multiple files in a folder like Generated\Folder1\Folder2.

**Comment by msawczyn** [comment]
Odd. You shouldn't be able to pick project folders that don't exist. How are you specifying those?

**Comment by dcastenholz** [comment]
They were initially created and then selected i the UI, but then I sometimes like to delete the whole Generated folder tree for a clean build.

**Comment by msawczyn** [comment]
Aha! A loophole, eh? Fair enough... :-)

**Comment by msawczyn** [comment]
Fixed in v1.3.0.4

**Comment by msawczyn** [comment]
Closing since fix was released.
